Fix issue #6

Ignagx opened the pull request:
Close #6

Ignagx committed:
Agrega constructor de copia

Ignagx committed:
Permite que cada cliente tenga mas de un vehiculo

Se cambia el objeto vehicle a vehicleVector,
siendo un vector, el cliente puede tener mas de un
vehiculo. Ademas, agrega los metodos addVehicle
y removeVehicle, permitiendo la permutacion del
vector.

Ignagx commented:
Falta implementar al pedir los datos de un cliente, que complete con los datos de todos lo vehiculos

Corte0 commented:
habría que implementar un método para eliminar el Vehículo por la patente además de por el índice.

Corte0 commented on Vehicle.h:
```diff
@@ -16,6 +16,7 @@ friend std::ostream& operator<<(std::ostream& os, const Vehicle& vehicle);
     std::string model;
 
 public:
+    Vehicle();
```
Esto para que se agregó?

Ignagx replied:
Es el constructor de copia. Habiamos hablado que en el metodo de vectores push_back(), se crea un nuevo objeto que se agrega al vector. Necesitabamos el constructor de copia para que se pueda hacer eso.

Corte0 replied:
eso no es un constructor de copia
el constructor de copia deberia recibir un objeto vehicle por referencia
no esta haciendo nada esa linea para mi

Ignagx replied:
Ok, pense que eran equivalentes, ahi borre la linea y no hacía nada

Corte0 commented:
Otra cosa, estos cambios hacen que la clase DataBase no compile. tenes que modificar las clases que dependen de esta para que funcionen tambien

Ignagx committed:
Borra el constructor default

Se habia confundido el constructor de copia con el default. Al final esa linea no hacia nada.

Ignagx committed:
Crea metodo para eliminar vehiculos por license

Se cambia el nombre de removeVehicle a removeVehicleByLicense

Ignagx committed:
Agrega valor int devuelto en removeVehicleByPos

Se agrega que el metodo devuelva un valor de tipo entero para deteccion de errores

Corte0 commented on Client.h:
```diff
@@ -28,7 +28,7 @@ friend std::ostream& operator<<(std::ostream& os, const Client& Client);
 
     void setId(unsigned long long id);
     void addVehicle(const Vehicle& vehicle);
-    void removeVehicleByPos(int vehiclePos);
```
me gusta que devuelva un valor, le agregaría que devuelva un codigo de error de vehiculo no encontrado, deberias agregar el codigo de derror a Error.h.
tambien deberias implementar lo mismo en el metodo que obtiene por patente

Ignagx replied:
Doing

Ignagx replied:
Done

Ignagx committed:
Evita que se creen vehiculos vacios al instanciar

Ignagx committed:
Permite el manejo de varios vehiculos por cliente

Crea el metodo addMultipleVehicles() para agregar varios vehiculos a un cliente
Ademas, modifica los metodos getClientByName() y getClientById para que pueda
devolver apropiadamente la el objeto con todos los vehiculos

Ignagx committed:
Maneja error de vehiculo inexistente al borrar

Ignagx committed:
Borra archivos comiteados por accidente

Corte0 commented on Client.cpp:
```diff
@@ -36,7 +37,7 @@ void Client::addVehicle(const Vehicle& vehicle) {
+  int vehicleFound = CL_VEHICLE_NOT_FOUND;
   if(vehiclePos > 0 && vehiclePos <= vehicleVector.size()){
     vehicleVector.erase(vehicleVector.begin() + vehiclePos-1);
     vehicleFound = 1;
```
Debería devolver 0 si no hay error

Ignagx replied:
okok ahi lo cambio

Corte0 commented on Error.h:
```diff
@@ -24,7 +25,8 @@ const std::string errLib[][99] = {
     },
     {
         //CLIENT_ERROR  [100 - 199]
```
Estos errores son del otro cliente, el que esta haciendo gaston, no de la clase. Esta feo ? si
pero tampoco se en que categoria pondria el error.

Ignagx replied:
Pero lo cambio, lo dejo asi? que lo resuelva gaston jkasjas

Corte0 commented on main.cpp:
che culia

Corte0 commented on SAPEI.db:
why, just why.

Corte0 commented on miPrograma:
T-T

Corte0 reviewed (commented):
nais

Ignagx committed:
Cambia valor devuelto cuando no hay error

gaston1709 reviewed (commented):
Saca el SAPEI.db del commit

gaston1709 reviewed (changes requested):
sacar el SAPEI.db del commit

Ignagx committed:
Borra Sapei.db, agregado por accidente

Ignagx committed:
Revierte cambios de manejo de errores

Se hacen estos cambios en fix-issue-#30

Corte0 commented on Client.cpp:
```diff
@@ -18,27 +23,42 @@ double Client::getBalance() const {
 }
 
 std::ostream& operator<<(std::ostream& os, const Client& client) {
+    std::string vehiculos;
```
Esta variable no se usa

Corte0 commented on DataBase.cpp:
```diff
@@ -287,31 +293,42 @@ Client DataBase::getClientById(int id) {
+            string model(reinterpret_cast<const char*>(sqlite3_column_text(stmt, 10)));
+
+            Vehicle vehicle(license, type, color, brand, model);
+            client.addVehicle(vehicle);  // Assuming Client has an addVehicle() method
```
Borrar este comentario

Corte0 committed:
Elimina variabel no utilizada

Corte0 committed:
Elimina comentario de dudosa procedencia
